docs/eks-installation.md

```diff
@@ -117,6 +121,127 @@ Loading locally built `antrea/nephe` images to EKS cluster.
 ~/terraform/eks destroy
 ```
 
+## Setup AWS IAM Role Based Access [Optional]
+
+In addition to using AWS credentials for access, Nephe provides support for IAM
+role-based access, offering an alternative method to grant access and
+permissions. IAM role-based access allows for fine-grained control, simplified
+access management and improved security by eliminating the need to manage
+individual AWS credentials. However, the limitation of IAM role-based access is
+that it can only be used when Nephe is deployed on an EKS cluster.
+
+This documentation section will guide you through the setup process in AWS for
+IAM role-based access for Nephe.
+
+### Create an IAM Policy
+
+Navigate to the IAM dashboard on AWS console and select policies.
+Create a policy with the following permissions.
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [{
+    "Effect": "Allow",
+    "Action": [
+      "ec2:AuthorizeSecurityGroupEgress",
+      "ec2:AuthorizeSecurityGroupIngress",
+      "ec2:CreateSecurityGroup",
+      "ec2:CreateTags",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteTags",
+      "ec2:DescribeInstances",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DescribeSecurityGroups",
+      "ec2:DescribeVpcs",
+      "ec2:DetachInternetGateway",
+      "ec2:ModifyNetworkInterfaceAttribute",
+      "ec2:ModifySecurityGroupRules",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RevokeSecurityGroupIngress",
+      "ec2:UpdateSecurityGroupRuleDescriptionsEgress",
+      "ec2:UpdateSecurityGroupRuleDescriptionsIngress"
+    ],
+    "Resource": "*"
+  }]
+}
+```
+
+### Create an IAM Role using AWS Console
+
+Navigate to the IAM dashboard on AWS console and select roles.
+
+Create an IAM role and select `AWS account` for trusted entity type. The
+external id option is not required but recommended for improved security. It can
+be any string but common practice is to generate an UID for Nephe. The AWS
+account number should be the account where Nephe EKS cluster is deployed.
+
+- Alternatively for the `custom trust policy` option, you can use the following.
```

Comment:
Could you update this in such a way, we create an IAM Role with some name xyz with example, and use that xyz in CPA secret?

Reply:
Not really, we don't know user's AWS account id, so there will be a variable anyway.